### PR TITLE
[Prim][PIR] group_norm decomposite rule support dynamic shape

### PR DESCRIPTION
**Pull request by zeroRains**
### PR types

New features
### PR changes

Others
### Description

support dynamic shape for group_norm

### CODE REVIEW

File: test/prim/pir_prim/test_prim_sub_graph_dynamic_shape.py [hunk 1]
```diff
@@ -365,5 +394,57 @@ def setUp(self):
...
 
+class TestPrimGroupNorm1(unittest.TestCase):
+    def setUp(self):
+        np.random.seed(2023)
+        self.dtype = "float32"
+        self.x_shape = [50, 640, 10, 20]
+        self.init_x_shape = [None, 640, None, None]
+        self.x = np.random.random(self.x_shape).astype(self.dtype)
+        self.net = group_norm_net1
+        self.necessary_ops = "pd_op.flatten"
```
Comment (cyber-pioneer) [inline test/prim/pir_prim/test_prim_sub_graph_dynamic_shape.py]: group_norm

Comment (zeroRains) [reply test/prim/pir_prim/test_prim_sub_graph_dynamic_shape.py]: Done

File: paddle/fluid/primitive/composite/composite.h [hunk 2]
```diff
@@ -890,21 +890,38 @@ std::tuple<Tensor, Tensor, Tensor> group_norm_decomp(
...
-  auto out = reshape<T>(res, x_dim);
-
+  Tensor out, mean_, var_;
+  if (has_dynamic_shape(x.shape())) {
+    Tensor x_dim = shape<T>(x);
+    std::vector<int64_t> one_axis(1, 1);
+    Tensor x_shape = get_slice<T>(x_dim, 0) * groups;
+    Tensor dim_1 = full<T>({1}, -1, x_dim.type());
+    x_shape = concat<T>({x_shape, dim_1});
+    x_cast = backend::reshape<T>(x_cast, x_shape);
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/composite/composite.h:900]:
```suggestion
    x_cast = reshape<T>(x_cast, x_shape);
```

File: paddle/fluid/primitive/composite/composite.h [hunk 3]
```diff
@@ -890,21 +890,38 @@ std::tuple<Tensor, Tensor, Tensor> group_norm_decomp(
...
+    mean_ = mean_decomp<T>(x_cast, IntArray(one_axis), true);
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/composite/composite.h:901]: remove IntArray

**Review by cyber-pioneer — APPROVED** [review]

**Pull request merged**